pkgs/applications/science/electronics/kicad/libraries.nix

```diff
@@ -24,9 +26,16 @@ let
           zip
         ];
 
-      postInstall = lib.optional (name == "packages3d") ''
-        find $out -type f -name '*.step' | parallel 'stepreduce {} {} && zip -9 {.}.stpZ {} && rm {}'
-      '';
+      postInstall = lib.optionalString (name == "packages3d") (
+        lib.concatStringsSep "\n" (
+          lib.optional (with3dCompressed && !with3d) ''
+            find $out -type f -name '*.step' | parallel 'stepreduce {} {} && zip -9 {.}.stpZ {} && rm {}'
```

Comment:
Could you explain the logic on this line here, please? Why are you compressing the files, only to then delete them?

Reply:
Oh and also, the package `kicad.libraries.packages3d` is always included when `with3d` is `true`, so there is no need to check it here.

Reply:
Hi, thanks for the review!

This line will do compression to `.stpZ` and then remove the `.step` file, it is the original line used today.
What I do is that I added a line that does not do the `rm` in case you want both `.stpZ` and `.step`.

An improvement to be made is to still do `stepreduce` when `with3dCompressed = false` and `with3d = true`.

Reply:
Since there has not been any more feedback here I'm marking this as resolved.